Source/WebCore/html/track/TextTrackCueList.cpp

```diff
@@ -129,26 +129,26 @@ void TextTrackCueList::clear()
 
 void TextTrackCueList::updateCueIndex(const TextTrackCue& cue)
 {
-    auto cuePosition = m_vector.begin() + cueIndex(cue);
-    auto afterCuePosition = cuePosition + 1;
-
-    ASSERT_SORTED(m_vector.begin(), cuePosition);
-    ASSERT_SORTED(afterCuePosition, m_vector.end());
-
-    auto reinsertionPosition = std::upper_bound(m_vector.begin(), cuePosition, *cuePosition, cueSortsBefore);
-    if (reinsertionPosition != cuePosition)
-        std::rotate(reinsertionPosition, cuePosition, afterCuePosition);
+    auto vectorSpan = m_vector.mutableSpan();
+    auto cueIndex = this->cueIndex(cue);
+    auto untilCuePosition = vectorSpan.first(cueIndex);
+    auto cuePosition = vectorSpan.subspan(cueIndex).begin();
+    auto afterCuePosition = vectorSpan.subspan(cueIndex + 1);
```

Comment:
Now that these are spans I find the word `position` in these variable names actively confusing. Like how can a position be sorted? And nothing in `cuePosition` makes it clear that it's the cue and everything after it. Could clarify with 3 better names. Maybe `valuesBeforeCue`, `cueAndValuesAfterCue`, and `valuesAfterCue`, but I hope we can come up with shorter better names.

But I'm not sure, maybe we can just leave these as is.

Reply:
Yes, I agree Position is confusing. I have renamed the variables.